### ISSUE TITLE
Can't sync from 0 on devnet

### ISSUE THREAD

**Issue by samharperpittam** [ArkEcosystem/core#4167]
Since the release of `3.0.0-next.9`, nodes can no longer sync from 0 due to no peers being available.

Log below:
```
1|ark-relay  | [2020-11-12 15:33:50.702] INFO	: Starting Round 1
1|ark-relay  | [2020-11-12 15:33:50.713] DEBUG	: Loaded 51 active delegates
1|ark-relay  | [2020-11-12 15:33:50.718] INFO	: Saving round 1
1|ark-relay  | [2020-11-12 15:33:50.743] INFO	: Your network connectivity has been verified by 208.67.220.220
1|ark-relay  | [2020-11-12 15:33:50.760] INFO	: Your NTP connectivity has been verified by time.google.com
1|ark-relay  | [2020-11-12 15:33:50.761] INFO	: Local clock is off by 2ms from NTP
1|ark-relay  | [2020-11-12 15:33:50.762] DEBUG	: GET https://raw.githubusercontent.com/ARKEcosystem/peers/master/devnet.json
1|ark-relay  | [2020-11-12 15:33:53.332] DEBUG	: Checking ports of 0 peer.
1|ark-relay  | [2020-11-12 15:33:53.333] INFO	: Checking 0 peers
1|ark-relay  | [2020-11-12 15:33:53.334] INFO	: 0 of 0 peers on the network are responsive
1|ark-relay  | [2020-11-12 15:33:53.334] INFO	: Median Network Height: 0
1|ark-relay  | [2020-11-12 15:33:53.335] DEBUG	: GET https://raw.githubusercontent.com/ARKEcosystem/peers/master/devnet.json
1|ark-relay  | [2020-11-12 15:33:55.473] INFO	: Couldn't find enough peers. Falling back to seed peers.
1|ark-relay  | [2020-11-12 15:33:56.490] DEBUG	: event 'STARTED': "init" -> {"syncWithNetwork":"syncing"} -> actions: [checkLastDownloadedBlockSynced]
1|ark-relay  | [2020-11-12 15:33:56.491] DEBUG	: Queued chunks of blocks (process: 0)
1|ark-relay  | [2020-11-12 15:33:56.492] DEBUG	: event 'NOTSYNCED': {"syncWithNetwork":"syncing"} -> {"syncWithNetwork":"downloadBlocks"} -> actions: [downloadBlocks]
1|ark-relay  | [2020-11-12 15:33:56.494] ERROR	: Could not download blocks: we have 0 peers
1|ark-relay  | [2020-11-12 15:33:56.494] INFO	: Could not download any blocks from any peer from height 2
1|ark-relay  | [2020-11-12 15:33:56.495] DEBUG	: event 'NOBLOCK': {"syncWithNetwork":"downloadBlocks"} -> {"syncWithNetwork":"syncing"} -> actions: [checkLastDownloadedBlockSynced]
1|ark-relay  | [2020-11-12 15:33:56.495] DEBUG	: Queued chunks of blocks (process: 0)
1|ark-relay  | [2020-11-12 15:33:56.496] DEBUG	: event 'SYNCED': {"syncWithNetwork":"syncing"} -> {"syncWithNetwork":"downloadFinished"} -> actions: [downloadFinished]
1|ark-relay  | [2020-11-12 15:33:56.497] INFO	: Block download finished
1|ark-relay  | [2020-11-12 15:33:56.497] DEBUG	: event 'PROCESSFINISHED': {"syncWithNetwork":"downloadFinished"} -> {"syncWithNetwork":"processFinished"} -> actions: [checkLastBlockSynced]
1|ark-relay  | [2020-11-12 15:33:56.498] DEBUG	: event 'SYNCED': {"syncWithNetwork":"processFinished"} -> {"syncWithNetwork":"end"} -> actions: [syncingComplete]
1|ark-relay  | [2020-11-12 15:33:56.499] INFO	: Blockchain 100% in sync
1|ark-relay  | [2020-11-12 15:33:56.499] DEBUG	: event 'SYNCFINISHED': {"syncWithNetwork":"end"} -> "idle" -> actions: [checkLater, blockchainReady]
1|ark-relay  | [2020-11-12 15:33:57.493] INFO	: Checking 0 peers
1|ark-relay  | [2020-11-12 15:33:57.495] INFO	: Public API (HTTP) Server started at http://0.0.0.0:4003
1|ark-relay  | [2020-11-12 15:34:55.474] DEBUG	: Checking ports of 0 peer.
1|ark-relay  | [2020-11-12 15:34:55.475] INFO	: Checking 0 peers
1|ark-relay  | [2020-11-12 15:34:55.477] DEBUG	: GET https://raw.githubusercontent.com/ARKEcosystem/peers/master/devnet.json
1|ark-relay  | [2020-11-12 15:34:57.288] INFO	: Couldn't find enough peers. Falling back to seed peers.
1|ark-relay  | [2020-11-12 15:34:58.226] DEBUG	: Throttling outgoing requests to 167.114.29.52/p2p.blocks.getBlocks to avoid triggering their rate limit
1|ark-relay  | [2020-11-12 15:35:57.290] DEBUG	: Checking ports of 0 peer.
1|ark-relay  | [2020-11-12 15:35:57.291] INFO	: Checking 0 peers
1|ark-relay  | [2020-11-12 15:35:57.292] DEBUG	: GET https://raw.githubusercontent.com/ARKEcosystem/peers/master/devnet.json
1|ark-relay  | [2020-11-12 15:35:59.495] INFO	: Couldn't find enough peers. Falling back to seed peers.
1|ark-relay  | [2020-11-12 15:36:59.497] DEBUG	: Checking ports of 0 peer.
1|ark-relay  | [2020-11-12 15:36:59.498] INFO	: Checking 0 peers
1|ark-relay  | [2020-11-12 15:36:59.498] DEBUG	: GET https://raw.githubusercontent.com/ARKEcosystem/peers/master/devnet.json
1|ark-relay  | [2020-11-12 15:37:01.545] INFO	: Couldn't find enough peers. Falling back to seed peers.
1|ark-relay  | [2020-11-12 15:37:02.458] DEBUG	: Throttling outgoing requests to 167.114.29.51/p2p.blocks.getBlocks to avoid triggering their rate limit
1|ark-relay  | [2020-11-12 15:37:02.480] DEBUG	: Throttling outgoing requests to 167.114.29.52/p2p.blocks.getBlocks to avoid triggering their rate limit
1|ark-relay  | [2020-11-12 15:37:02.484] DEBUG	: Throttling outgoing requests to 167.114.29.55/p2p.blocks.getBlocks to avoid triggering their rate limit
1|ark-relay  | [2020-11-12 15:37:02.499] DEBUG	: Throttling outgoing requests to 167.114.29.54/p2p.blocks.getBlocks to avoid triggering their rate limit

**Comment by alessiodf** [comment]
FYI this is because the peer verifier only downloads block headers, not their transactions - presumably for efficiency reasons - but then that means `Blocks.BlockFactory.fromData` in `verifyPeerBlock` fails whenever it tries to verify non-empty blocks due to the missing transactions.

**Comment by air1one** [comment]
Exactly @alessiodf working on it now, so validation in `Blocks.BlockFactory.fromData` fails because even though we ask blocks with `headersOnly`, the new ser/deser with protobuf deserializes into an empty array transactions field : `transactions: []`. Which then triggers the validation error in `fromData`.
I'm adapting it so that when we ask `headersOnly` we don't have `transactions` field set at all.

**Comment by air1one** [comment]
Fixed by #4169 (not released yet, pull from develop to test it).
